### ISSUE TITLE
Questions about the training code

### ISSUE THREAD

**Issue by blackblue9** [uclaml/SPPO#6]
Thank you for sharing your code and making it open source.

But I noticed that the training code you used still uses the DPO trainer, which seems inconsistent with the SPPO in the paper. Can you explain why you use DPO instead of the suggested SPPO in your code?

**Comment by angelahzyuan** [comment]
We did not rename the Trainer because one version of our code has already been integrated into trl library under DPOTrainer with the loss_type "sppo_hard."

In our code, we specify the loss_type as "sppo" to represent SPPO loss.
